app/src/main/java/com/kickstarter/libs/utils/ApplicationLifecycleUtil.java

```diff
@@ -55,6 +61,16 @@ public void onActivityResumed(final @NonNull Activity activity) {
         .compose(Transformers.neverError())
         .subscribe(c -> this.config.config(c));
 
+      // Refresh the user
+      final Observable<User> user = this.currentUser.observable();
+
+      final String accessToken = this.currentUser.getAccessToken();
+
+      Observable.just(accessToken)
+        .filter(ObjectUtils::isNotNull)
+        .zipWith(user, Pair::create)
+        .subscribe(tokenUserPair -> this.currentUser.refresh(tokenUserPair.second));
```

Comment:
Hey as far as I can tell (but I might be missing something in the Java) is that `this.currentUser.refresh()` will just refresh with whatever is passed to it. So I think this will just take the `currentUser` and "refresh" it again. At which point do we make a new call to the API to fetch a fresh user?

Also, in this particular scenario it might not even be necessary to use RxJava because we're just imperatively lifting the token into FRP land in order to `subscribe` to it (with the user) and then to refresh the user with the user 🤔 

I think I might need a little more insight into how the lifecycle stuff works in Android but maybe we should pair on this 💪

Reply:
😅 look for my calendar invite!

Reply:
Aah got it! I missed the _fetch_ part of that 😄 

Then actually this is probably fine but couldn't it just be:

```
this.client.fetchCurrentUser()
  .filter(accessToken != null) //pseudeocode
  .subscribe(freshUser -> this.currentUser.refresh(freshUser))
```

Think that just removes the need to lift the accessToken into RxJava and to zip it with the new user.